### PR TITLE
Feature/update endpoint to fetch tweets by user

### PR DESCRIPTION
**Pull request by alexandrtovmach**

### CODE REVIEW

File: src/repositories/tweet.repository.ts [hunk 1]
```diff
@@ -1,8 +1,18 @@
-import { TweetModel } from "../models/Tweet.model";
+import { FilterQuery, PaginateOptions } from "mongoose";
+
+import { Tweet, TweetModel } from "../models/Tweet.model";
+
+export const getList = async (
```
Comment (VitaliiDenysiuk) [inline src/repositories/tweet.repository.ts]:
```suggestion
export const getList =  (
```
no await

File: src/repositories/tweet.repository.ts [hunk 2]
```diff
@@ -34,12 +39,17 @@ export const updateOne = async (
...
+
 export const changeStats = async (
   id: string,
   fieldName: string,
   value: number
 ) => {
   return TweetModel.updateOne({ _id: id }, { $inc: { [fieldName]: value } });
 };
+
+export const deleteOne = async (id: string) => {
```
Comment (VitaliiDenysiuk) [inline src/repositories/tweet.repository.ts]:
```suggestion
export const deleteOne = (id: string) => {
```
no await

File: src/routes/bookmarks.route.ts [hunk 3]
```diff
@@ -10,26 +9,20 @@ import {
...
-  .route("/")
-  .post(isAuthorized, async (req: Request, res: Response) => {
-    const { tweet, owner } = req.body;
-    await addBookmark(tweet, owner);
-    res.sendStatus(200);
-  });
+bookmarksRouter.route("/").post(isAuthorized, async (req, res) => {
+  const { tweet, owner } = req.body;
+  await addBookmark(tweet, owner);
+  res.sendStatus(200);
```
Comment (VitaliiDenysiuk) [inline src/routes/bookmarks.route.ts]:
```suggestion
  const newBookmarks = await addBookmark(tweet, owner);
  return res.status(201).send(newBookmarks);
```

**Review by AndrewKaenko — APPROVED** [review]
Looks perfect to me

File: src/routes/likes.route.ts [hunk 4]
```diff
@@ -14,34 +12,26 @@ import {
...
-likesRoutes
-  .route("/")
-  .post(isAuthorized, async (req: Request, res: Response) => {
-    const user = req.user as User;
-    await changeLike(req.body.tweet, user._id);
-    res.sendStatus(200);
-  });
+likesRoutes.route("/").post(isAuthorized, async (req, res) => {
+  await changeLike(req.body.tweet, req.user._id);
+  res.sendStatus(200);
```
Comment (VitaliiDenysiuk) [inline src/routes/likes.route.ts]:
```suggestion
    const newLike = await changeLike(req.body.tweet, req.user._id);
  res.status(201).send(newLike);
```

File: src/routes/users.route.ts [hunk 5]
```diff
@@ -12,48 +11,38 @@ import {
...
-  .put(isAuthorized, async (req: Request, res: Response) => {
-    const result = await updateUser(req.params.id, req.body);
+  }
+  return res.sendStatus(500);
+});
+
+usersRoutes.route("/").post(isAuthorized, async (req, res) => {
+  if (req.body.name) {
+    const result = await createUser(req.body);
     return res.send(result);
```
Comment (VitaliiDenysiuk) [inline src/routes/users.route.ts]:
```suggestion
    return res.status(201).send(result);
```

File: src/services/bookmarks.service.ts [hunk 6]
```diff
@@ -6,11 +7,9 @@ import {
   listBookmarks,
 } from "../repositories/bookmarks.repository";
 
-export const getBookmarks = async (
-  data: PaginationParameters<never, never>
-) => {
-  const bookmarksList = await listBookmarks(...data.get());
-  return bookmarksList;
+export const getBookmarks = async (req: Request) => {
```
Comment (VitaliiDenysiuk) [inline src/services/bookmarks.service.ts]:
```suggestion
export const getBookmarks =  (req: Request) => {
```
no await

File: src/services/bookmarks.service.ts [hunk 7]
```diff
@@ -6,11 +7,9 @@ import {
...
+  const [query, options] = new PaginationParameters(req).get();
+  return listBookmarks(query, options);
 };
 
 export const addBookmark = async (owner: string, tweet: string) => {
```
Comment (VitaliiDenysiuk) [inline src/services/bookmarks.service.ts]:
```suggestion
export const addBookmark =  (owner: string, tweet: string) => {
```
no await

File: src/services/followers.service.ts [hunk 8]
```diff
@@ -6,11 +7,9 @@ import {
   listFollowers,
 } from "../repositories/followers.repository";
 
-export const getFollowers = async (
-  query: PaginationParameters<never, never>
-) => {
-  const followersList = await listFollowers(...query.get());
-  return followersList;
+export const getFollowers = async (req: Request) => {
```
Comment (VitaliiDenysiuk) [inline src/services/followers.service.ts]:
```suggestion
export const getFollowers = (req: Request) => {
```
no await

File: src/services/tweets.service.ts [hunk 9]
```diff
@@ -9,6 +10,16 @@ import {
   updateOne,
 } from "../repositories/tweet.repository";
 
+export const getAllTweetsOfCurrentUser = async (req: Request) => {
```
Comment (VitaliiDenysiuk) [inline src/services/tweets.service.ts]:
```suggestion
export const getAllTweetsOfCurrentUser = (req: Request) => {
```
no await

File: src/services/tweets.service.ts [hunk 10]
```diff
@@ -9,6 +10,16 @@ import {
...
+  const [query, options] = new PaginationParameters(req).get();
+  return getList({ ...query, author: req.user._id }, options);
+};
+
+export const getAllTweets = async (req: Request) => {
```
Comment (VitaliiDenysiuk) [inline src/services/tweets.service.ts]:
```suggestion
export const getAllTweets = (req: Request) => {
```
no await

File: src/services/tweets.service.ts [hunk 11]
```diff
@@ -48,3 +51,7 @@ export const changeTweetStats = async (
 ) => {
   return changeStats(id, fieldName, value);
 };
+
+export const deleteTweet = async (id: string) => {
```
Comment (VitaliiDenysiuk) [inline src/services/tweets.service.ts]:
```suggestion
export const deleteTweet = (id: string) => {
```
no await

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]

**Review by fominmaksim — APPROVED** [review]

**Review by VitaliiDenysiuk — APPROVED** [review]

**Pull request merged**